app/src/main/java/io/intrepid/skeleton/base/BaseMvpActivity.java

```diff
@@ -0,0 +1,50 @@
+package io.intrepid.skeleton.base;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.annotation.CallSuper;
+import android.support.annotation.NonNull;
+
+public abstract class BaseMvpActivity<T extends BasePresenter> extends BaseActivity implements BaseView {
+
+    protected T presenter;
+
+    @NonNull
+    public abstract T createPresenter(PresenterConfiguration configuration);
+
+    protected abstract int getLayoutResourceId();
+
+    @CallSuper
+    protected void onViewCreated(Bundle savedInstanceState) {
+        PresenterConfiguration configuration = getSkeletonApplication().getPresenterConfiguration();
+        presenter = createPresenter(configuration);
+        presenter.onViewCreated();
+    }
+
+    @Override
+    @CallSuper
+    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
+        super.onActivityResult(requestCode, resultCode, data);
+        presenter.bindView(this);
```

Comment:
So in the case that activity A launches activity B with startActivityForResult, there was a crash with a null pointer exception returning to Activity A on our view interface. This is simply a guard against that, and if it can be removed, it should be deleted.